### DISCUSSION TITLE
patching firefox for D50 chad with negative XYZ values on Linux and Windows

### DISCUSSION THREAD

**Discussion by 24fpsDaVinci** [saucecontrol/Compact-ICC-Profiles#11] [Show and tell]
on linux and windows you can patch firefox's icc parsing here gfx/qcms/src/iccread.rs around line 400

```    
if !cfg!(target_os = "macos") {
        negative = (rX < 0.)
            || (rY < 0.)
            || (rZ < 0.)
            || (gX < 0.)
            || (gY < 0.)
            || (gZ < 0.)
            || (bX < 0.)
            || (bY < 0.)
            || (bZ < 0.);
    } else {
        // Chromatic adaption to D50 can result in negative XYZ, but the white
        // point D50 tolerance test has passed. Accept negative values herein.
        // See https://bugzilla.mozilla.org/show_bug.cgi?id=498245#c18 onwards
        // for discussion about whether profile XYZ can or cannot be negative,
        // per the spec. Also the https://bugzil.la/450923 user report.

        // FIXME: allow this relaxation on all ports?
        negative = false; // bogus
    }
    if negative {
        return true;
    }
```

I already requested this on bugzilla but for reasons i cant comprehend firefox devs dont want to enable this on windows and linux builds. maybe you can include this in the readme.

**Comment by jrmuizel** [comment]
What's the bug number?

**Comment by 24fpsDaVinci** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1792469

**Comment by saucecontrol** [comment]
Interesting.  I noticed a while back that Chrome was fixed and meant to update the compatibility note here.

Moving this over to Discussions since it's not actionable here.

**Reply by 24fpsDaVinci** [reply]
are you referring to force P3-D65?

**Reply by saucecontrol** [reply]
Is that a Chrome option?  I only see that images with Display P3 display correctly in Chrome now.  If that's a specific override that makes P3 work specifically but not other profiles with negative primary values, I don't know.

**Comment by jrmuizel** [comment]
This should be fixed in Firefox 108

**Reply by jrmuizel** [reply]
That's surprising to me. Do you have a test case that you can share?

**Comment by 24fpsDaVinci** [comment]
just adding another issue in here, it appears firefox only loads matrix profiles, luts are completely skipped, at least with iccv2 profiles generated by argyllcms. To test, use a debug profile where the matrix colors are inverted to see if matrix or lut is being loaded.

https://www.argyllcms.com/doc/colprof.html#a
